src/command_modules/azure-cli-sql/azure/cli/command_modules/sql/custom.py

```diff
@@ -2071,3 +2075,161 @@ def managed_db_restore(
         managed_instance_name=target_managed_instance_name,
         resource_group_name=target_resource_group_name,
         parameters=kwargs)
+
+###############################################
+#              sql failover-group             #
+###############################################
+
+
+# pylint: disable=too-few-public-methods
+class FailoverPolicyType(Enum):
+    automatic = 'Automatic'
+    manual = 'Manual'
+
+
+def failover_group_create(
+        cmd,
+        client,
+        resource_group_name,
+        server_name,
+        failover_group_name,
+        partner_server,
+        partner_resource_group=None,
+        failover_policy=FailoverPolicyType.automatic.value,
+        grace_period=1,
+        add_db=None):
+
+    from six.moves.urllib.parse import quote  # pylint: disable=import-error
+    from azure.cli.core.commands.client_factory import get_subscription_id
+
+    # Build the partner server id
+    partner_server_id = "/subscriptions/{}/resourceGroups/{}/providers/Microsoft.Sql/servers/{}".format(
+        quote(get_subscription_id(cmd.cli_ctx)),
+        quote(partner_resource_group or resource_group_name),
+        quote(partner_server))
+
+    partner_server = PartnerInfo(id=partner_server_id)
+
+    # Convert grace period from hours to minutes
+    grace_period = int(grace_period) * 60
```

Comment:
Why is grace period in hours for create, but the underlying api (and therefore the json response from the command) will be in seconds? If you want to allow hours to be specified, you can use `SizeWithUnitConverter` from params.py with unit_map like `dict(s = 1, h = 60)`, this will allow default to be in seconds but allows user to specify `--grace-period 5h` for example.

Reply:
The decision to have grace period in hours was made already after ARM API was published. Hence the discrepancy there. However, we are now unable to support time shorter than 1 hour anyway, so both Powershell and CLI should support only this value. I am hesitant to make this with unit marker, because the natural way would be for customer to try having grace period smaller than 1 hour.